### PR TITLE
Added preRender callback

### PR DESCRIPTION
**Pull request by SathishGovindaraju**
- Callback function which gets called before the component gets rendered

### CODE REVIEW

**Review by coderas — APPROVED** [review]
joy.

Non-breaking change 👍

File: README.md [hunk 1]
```diff
@@ -67,6 +67,14 @@ For example:
     // Other callbacks
   }));  
 ```
+### `preRender`
+
+Callback function called before rendering `App`. This callback should return a valid component.
```
Comment (afenton90) [inline README.md]: Just need to make it clear here that this component can be resolved in a `Promise`

**Review by afenton90 — CHANGES_REQUESTED** [review]
One line comment then this is ready to go 👍

**Review by afenton90 — APPROVED** [review]
Great work! 💯 
Could I request that an example is added in a future PR.

**Pull request merged**